controllers/users.js

```diff
@@ -5,17 +5,90 @@ const uuid = require('uuid');
 const bcrypt = require('bcrypt');
 const parse = require('co-body');
 
+const loadConfig = require('./load-config');
+let fbOptions = loadConfig.DEFAULTS;
+
+const FB = require('fb');
+let promise = FB;
+const fb = FB.extend({appId: fbOptions.appId, appSecret: fbOptions.appSecret});
+
 const User = require('../models').models.User;
 
 exports.login = function* (next) {
+  let ctx = this;
+
+  let fbToken = this.request.body.fbToken;
+  fb.setAccessToken(fbToken);
+
+  let p1 = new Promise(function(resolve, reject){
+    fb.api(fbOptions.apiURL, function(res){
+      if (!res || res.error) {
+        console.log(!res ? 'error occurred' : res.error);
+        reject(res.error);
+      }
+      resolve(res);
+    });
+  });
+
+  let fbData = yield p1
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => {
+      console.log('An error has returned: ' + err);
+  });
+
+  yield User.findOne({ 'profileInfo.fbId': fbData.id })
+    .then((user)=>{
+      if (user !== null) {
+        ctx.status = 200;
+        ctx.body = {
+          fbToken: user.fbToken,
+          authToken: user.authToken,
+          fbId:user.profileInfo.fbId,
+          firstName:user.profileInfo.firstName,
+          lastName:user.profileInfo.lastName,
+          picture:user.profileInfo.picture,
+          email:user.profileInfo.email
+        };
+      }
+    }).then((user)=>{
+      console.log('User is not in the database we need to create...');
+      let newToken = uuid.v4();
+      let newDocument = {
+        fbToken:'1234',     // fbToken
+        authToken:newToken,
+        profileInfo: {
+          fbId: fbData.id,            // fbData.id
+          firstName:'Percy',       // fbData.first_name
+          lastName:'Jackson',        // fbData.last_name
+          picture:'Lightening Thief Image',         // fbData.picture.data.url
+          email:'zues@pj.com'           // fbData.email
+        }
+      }
 
+      //  convert to a promise
+      let newUser = new User(newDocument);
+      newUser.save();
+      console.log('New user has been created!');
+      ctx.body = {
```

Comment:
Second time using the same structure. Write a function `serializeUser` or `User.prototype.serialized()` that acts like a serializer. It should take a user and returns an object with this format.